warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryPropertyMarkerSourceConsolidator.java

```diff
@@ -0,0 +1,42 @@
+package datawave.query.jexl.visitors;
+
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.JexlNodeFactory;
+import datawave.query.jexl.nodes.QueryPropertyMarker;
+import org.apache.commons.jexl2.parser.ASTAndNode;
+import org.apache.commons.jexl2.parser.JexlNode;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This visitor will ensure that query marker nodes always have a singular, root source node.
```

Comment:
Can we add something to indicate what is happening? "If a marker node has multiple sources they will be bundled within an ASTAndNode" or something like that

Reply:
Sure thing.